planseqlearn/psl/mp_env.py

```diff
@@ -97,115 +98,88 @@ def get_object_pose_mp(self, obj_idx=0):
         raise NotImplementedError
 
     def get_all_initial_object_poses(self):
-        self.initial_object_pos = [self.get_object_pose_mp(obj_idx=0)[0].copy()]
+        self.initial_object_pos_dict = {}
+        for obj_name, action in self.text_plan:
+            if action == "grasp":
+                self.initial_object_pos_dict[obj_name] = self.named_get_object_pose_mp(obj_name)[0].copy()
+
+    def get_curr_postcondition(self):
+        if self.text_plan[self.curr_plan_stage][1].lower() == "grasp":
+            return self.named_check_object_grasp(self.text_plan[self.curr_plan_stage][0])
+        elif self.text_plan[self.curr_plan_stage][1].lower() == "place":
+            return self.named_check_object_placement(self.text_plan[self.curr_plan_stage][0])
+        else:
+            raise NotImplementedError("Currently only supporting grasp and place postconditions")
 
     def reset(self, get_intermediate_frames=False, **kwargs):
         self.num_high_level_steps = 0
         self.object_idx = 0
         # reset wrapped env
         obs = self._wrapped_env.reset(**kwargs)
-        self.post_reset_burn_in()
-
-        # initialize trajectory variables
-        self.num_high_level_steps = 0
+        if self.text_plan is None:
+            self.text_plan = self.get_hardcoded_text_plan()
+            print(f"Text plan: {self.text_plan}")
+        self.curr_plan_stage = 0
+        self.curr_postcondition = self.get_curr_postcondition()
 
+        self.post_reset_burn_in()
+        if hasattr(self, "reset_precompute_sam_poses") and self.use_sam_segmentation:
+            self.reset_precompute_sam_poses()
         # reset position
         self.reset_pos, self.reset_ori = self._eef_xpos.copy(), self._eef_xquat.copy()
         self.reset_qpos, self.reset_qvel = (
             self.sim.data.qpos.copy(),
             self.sim.data.qvel.copy(),
         )
-        self.placement_poses = self.get_placement_poses()
         self.get_all_initial_object_poses()
         self.update_controllers()
+        # go to initial position 
         target_pos, target_quat = self.get_target_pos()
         if self.teleport_instead_of_mp:
             self.set_robot_based_on_ee_pos(
-                target_pos.copy(),
-                target_quat.copy(),
+                target_pos,
+                target_quat,
```

Comment:
i think you want to keep the copy? unless there was a reason to remove it.

Reply:
i think within the get target function i'll make sure everything is returning a copy anyway, but this is because for some of the envs, i just set the target quat to None since i think then it chooses the optimal target quat to achieve the eef xpos, and i was getting an error when i call (None).copy()

Reply:
yeah make sure everything is copied correctly in the target function then.